lib/identity/client.js

```diff
@@ -67,6 +68,60 @@ class Client extends PartialClient {
     options.headers.Authorization = `Bearer ${token}`
     return fetch(url, options)
   }
+
+  async createSecret(secret) {
+    const trimmedName = secret.secretName.trim()
+    const trimmedValue = secret.secretValue.trim()
+    if (trimmedName === '') {
+      throw { message: 'Name cannot be empty' }
+    }
+    const re = new RegExp('^[a-zA-Z0-9-_]{1,30}$')
+    if (!re.test(trimmedName)) {
+      throw {
+        message: 'Secret name must contain alphanumeric characters, -, or _',
+      }
+    }
+    if (trimmedValue === '') {
+      throw { message: 'Value cannot be empty' }
+    }
+    const groupName = [`tozny.secret.${this.storage.config.clientId}`]
```

Comment:
This could also be a single string, and then on line 91 you can do [groupName] and then on line 97 not have to index into this array.

More importantly though, for group name lets have both the realm name and the secret type in the group name

`tozny.secret.{realmName}.{clientId}.{secretType}`

Reply:
I believe there is a limit of some kind on record-type names -- is there any chance with a namespace like this we'd start hitting it? I don't know what is is off the top of my head, but iirc, it is not that large.

Reply:
🤦 - oh, this is group name. Still, these keys are getting long, let's make sure we are working within our own limits for both group name and record type.